nixos/modules/system/activation/top-level.nix

```diff
@@ -159,6 +160,33 @@ in
       };
     };
 
+    system.nixosVersionJsonData = lib.mkOption {
```

Comment:
`Data` doesn't mean anything. I think it'd be fine to call this `nixosVersionJson` and have the json drv in a let binding instead of an option.

(Also I think it doesn't _have_ to be a separate drv, but that doesn't have to change either)